### DISCUSSION TITLE
Squid + zapret

### DISCUSSION THREAD

**Discussion by 8hrsk** [bol-van/zapret#984] [Q&A]
Поднимаю прокси сервер в docker на своей машине. У меня есть белый ip и доменное имя (условно example.ru)

Zapret в Docker я поднял, но вот как всё это дело сопрячь со Squid я понятия не имею. Работаю с iptables, в них у меня есть список доменов, запросы к котрым я хочу проксирать.

Буду рад каким-либо подсказкам <3

**Comment by bol-van** [comment]
squid никак не поможет со списком доменов. Времена plain http ушли.
squid разве что как прокси может работать для клиентов. Но с этим и tpws --socks справится.

Если так нужен доменный список, есть стандартное решение dnsmasq+ipset и заворот по ipset
Но nfqws и tpws и так умеют по хостлистам работать. Проблемы будут только при применении syndata и wssize

**Reply by 8hrsk** [reply]
Получается, если на время отойти от идеи с iptables и просто пропускать весь трафик через прокси контейнер, в котором запущен запрет, то всё должно работать?

**Reply by bol-van** [reply]
Смотря как он организован. Есть ли в нем отдельный network namespace, как он соединен с основным хостом, не урезаны ли caps

**Reply by 8hrsk** [reply]
Ну, вообще, я пока развернул вмё в одном контейнере. Запускается образ, в нём запускается запрет и хватает готовый конфиг, который я в него передаю в докерфайле. В этом же контейнере запускается squid, слушающий порт 3128.
Но, видимо, то ли из-за того, что я пытаюсь настроить iptables, когда это не сделать не выходит, то ли еще чего-то, ничего не работает и при подключении к прокси на устройстве просто ничего не грузит.

В любом случае, спасибо за совет. Буду дальше разбираться
